guidelines/js/vue/nuxt/guidelines-with-explanations.md

```diff
@@ -0,0 +1,184 @@
+- Define routes in the pages directory following the file-based routing convention
+- Store app-level functionality in the plugins directory
+- Place static files in the public directory (formerly static in Nuxt 2)
+- Use the server directory for server-side code (API routes, middleware)
+
+**Explanation:**
+- The `.nuxt` directory contains auto-generated files for your Vue application and should never be manually modified
+- The `.output` directory holds production build files generated by `nuxt build` and should not be modified when deploying
+- The `assets` directory is processed by build tools and is ideal for CSS, SCSS, fonts, and images that need processing
+- The `components` directory benefits from auto-import functionality, eliminating the need for manual imports in templates; nested components are imported with their directory prefix (e.g., `ButtonPrimary` for `components/Button/Primary.vue`)
+- The `composables` directory stores reusable logic with auto-import functionality; by default, only top-level composables are auto-imported, but this can be configured in `nuxt.config.js`
+- The `content` directory works with the Nuxt Content module to create a file-based CMS using markdown, JSON, YAML, and CSV files
+- The `layouts` directory provides structural templates for pages; each layout must have a single root element (not a `<slot>`) to enable transitions
+- The `middleware` directory supports three types: anonymous (defined in pages), named (in middleware directory), and global (with .global suffix)
+- The `pages` directory automatically generates routes based on file structure; creating `pages/about.vue` automatically creates the `/about` route
+- The `plugins` directory contains code that runs at application startup; plugins are auto-registered and should not be added to nuxt.config separately
+- The `public` directory (formerly `static` in Nuxt 2) is served at the server root and contains files that need to maintain their filenames
+- The `server` directory contains server-side code including API routes, middleware, plugins, and utilities
+
+## Pages and Routing
+- Use dynamic routes appropriately
+- Implement nested routes when logical
+- Use middleware for route guards
+- Leverage route validation with definePageMeta
+- Use route parameters for dynamic content
+
+**Explanation:**
+- Dynamic routes (e.g., [id].vue) allow for reusing page components with different data based on parameters
+- Nested routes help organize related pages and maintain UI consistency for sections of your application
+- Middleware provides a clean way to implement authentication, analytics, or other cross-cutting concerns
+- definePageMeta allows you to define metadata for pages including layout, middleware, and validation
+- Route parameters should be validated to prevent unexpected behavior and improve security
+
+## Components
+- Create reusable components in the components directory
+- Use props validation
```

Comment:
I'd say: 

* Use TypeScript for props
* Use defineModel instead of a manual implementation of custom v-model
* Use script setup (with TS by default)
* Use props destructuring instead of withDefaults